### DISCUSSION TITLE
How to configure eslint so that an error occurs when calling asyncFunction without await?

### DISCUSSION THREAD

**Discussion by shangHai-F** [eslint/eslint#15658] [Ideas]
Hello guys！How to warn when I forget to `await` an async function in Javascript?

**Comment by nzakas** [comment]
This can’t be done with standard ESLint because it requires type information that ESLint doesn’t track. You maybe able to achieve this with typescript-eslint.

**Comment by shangHai-F** [comment]
@nzakas Thank you, I agree, I plan to rewrite JavaScript to TypeScript.

**Reply by voxpelli** [reply]
@shangHai-F You do not need to rewrite your JS to TS to use type information. `typescript-eslint` can work on typed JS as well. See eg. https://github.com/voxpelli/types-in-js#how-to-use-types-in-javascript on how to set up types in JS

**Reply by shangHai-F** [reply]
Got it, Thanks!

**Comment by chiefmikey** [comment]
I believe [eslint-plugin-no-floating-promise](https://github.com/SebastienGllmt/eslint-plugin-no-floating-promise) provides this behavior. If using typescript then [typescript-eslint](https://typescript-eslint.io/rules/no-floating-promises/) has this rule baked in as well.
